packages/circuit-ui/components/Dialog/Dialog.tsx

```diff
@@ -186,11 +197,11 @@ export const Dialog = forwardRef<HTMLDialogElement, DialogProps>(
 
     const onPolyfillBackdropClick = useCallback(
       (event: MouseEvent) => {
-        if (preventClose) {
+        if (preventOutsideClickClose) {
           event.preventDefault();
         }
       },
-      [preventClose],
+      [preventOutsideClickClose],
     );
 
     useEffect(() => {
```

Comment:
I would split this `useEffect` into multiple, granular effects to improve performance:

- Registering the dialog with the polyfill. This should only happen once, on mount.
- Setting the animation duration as a custom property. This should re-run whenever the animation duration value changes (i.e. it shouldn't use the `animationDurationRef`)
- Adding the `keydown` event listener...
- ...and adding the `close` event listener. I'd separate the event listener effects since they are added under different conditions